### PR TITLE
Ship Operations Synthetic

### PR DESCRIPTION
**Pull request by QuickLode**
# About the pull request
This PR adds a ship oriented Synthetic who is playable if there is no Survivor Synthetic present.

Issue at hand was although Synthetics readily accept RP, they are not able to easily go out of their way to initiate it due to their responsibilities outside of 3 phases: Pre-drop, Surgery, Post-drop.

This is a push to better incentivize the availability of Synthetics to RP. Currently they are burdened with a multitude of responsibilities compared to other roles which are able to prioritize RP. This effectively bridges that gap significantly, and encourages RP whilst by having 1 Synthetic operate more in-line with the bridge officer / ship maintenance synthetic within the bounds of lore.

BIG Thanks to Drathek for the role authority code!!!

Great suggestion by Beagle!!

# Explain why it's good for the game

Conducive to RP availability and pretty healthy for immersion within lore. Does not change the overall amount of Synthetics.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

image

image

image

image

built ok 
</details>

# Changelog
:cl:
add: Adds the Ship Operations Synthetic. This ship-oriented Synthetic assists the USS Almayer with various tasks and improves the atmosphere of the vessel, they maintain the entire ship and act as its pilot. This role is enabled if there is no Survivor Synthetic present. Huge thanks to Drathek for roleauthority code!
add: Changes role selection color of WJ to be slightly darker to differentiate from Synthetics.
code: restructured lobby role selection menu to accomadate new role
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by Diegoflores31** [comment]
i genuinely dont see where is the RP incentive and what kind of implementation there is to prevent this from just being a powercreep ? they way i see it this just allows for synths to steal the very little amount of work and responsability that shipside roles have . 
surgery ? with the synth instead of the docs because faster surgery
chemline? nah go with synth and he will use research juice instead
need something build ? synth
req ? synth

File: code/game/jobs/slot_scaling.dm [hunk 1]
```diff
@@ -46,7 +46,7 @@
 	return job_slot_formula(playercount,20,1,2,4)
 
 /proc/synth_slot_formula(playercount)
-	return job_slot_formula(playercount,120,1,1,2)
+	return job_slot_formula(playercount,100,1,1,2)
```
Comment (TheGamerdk) [inline code/game/jobs/slot_scaling.dm:49]: Old code / undocumented?

Comment (Drulikar) [reply code/game/jobs/slot_scaling.dm:49]: Fix this

File: code/game/jobs/job/civilians/support/synthetic.dm [hunk 2]
```diff
@@ -50,3 +50,42 @@
 	name = JOB_SYNTH
 	icon_state = "syn_spawn"
 	job = /datum/job/civilian/synthetic
+
+
+// ship synthetic
+
+/datum/job/civilian/synthetic/ship
+	title = JOB_SHIP_SYNTH
+	total_positions = 1
```
Comment (Drulikar) [inline code/game/jobs/job/civilians/support/synthetic.dm:59]: `/datum/job/civilian/synthetic` must then be reduced to 1

File: code/__DEFINES/mode.dm [hunk 3]
```diff
@@ -129,10 +129,11 @@
...
+#define ROLES_OFFICERS list(JOB_CO, JOB_XO, JOB_SO, JOB_AUXILIARY_OFFICER, JOB_INTEL, JOB_CAS_PILOT, JOB_DROPSHIP_PILOT, JOB_TANK_CREW, JOB_DROPSHIP_CREW_CHIEF, JOB_SEA, JOB_CORPORATE_LIAISON, JOB_COMBAT_REPORTER, JOB_SYNTH, JOB_SHIP_SYNTH, JOB_CHIEF_POLICE, JOB_WARDEN, JOB_POLICE)
 GLOBAL_LIST_INIT(ROLES_CIC, list(JOB_CO, JOB_XO, JOB_SO, JOB_WO_CO, JOB_WO_XO))
 GLOBAL_LIST_INIT(ROLES_CIC_ANTAG, list(JOB_UPP_SRLT_OFFICER, JOB_UPP_KPT_OFFICER, JOB_UPP_CO_OFFICER, JOB_UPP_COMMISSAR))
 GLOBAL_LIST_INIT(ROLES_AUXIL_SUPPORT, list(JOB_AUXILIARY_OFFICER, JOB_INTEL, JOB_CAS_PILOT, JOB_DROPSHIP_PILOT, JOB_DROPSHIP_CREW_CHIEF, JOB_TANK_CREW, JOB_WO_CHIEF_POLICE, JOB_WO_SO, JOB_WO_CREWMAN, JOB_WO_POLICE, JOB_WO_PILOT))
+GLOBAL_LIST_INIT(ROLES_MISC, list(JOB_SYNTH, JOB_SHIP_SYNTH, JOB_WORKING_JOE, JOB_SEA, JOB_CORPORATE_LIAISON, JOB_COMBAT_REPORTER, JOB_MESS_SERGEANT, JOB_WO_CORPORATE_LIAISON, JOB_WO_SYNTH))
 GLOBAL_LIST_INIT(ROLES_AUXIL_SUPPORT_ANTAG, list(JOB_UPP_CREWMAN))
 GLOBAL_LIST_INIT(ROLES_MISC, list(JOB_SYNTH, JOB_WORKING_JOE, JOB_SEA, JOB_CORPORATE_LIAISON, JOB_COMBAT_REPORTER, JOB_MESS_SERGEANT, JOB_WO_CORPORATE_LIAISON, JOB_WO_SYNTH))
```
Comment (Drulikar) [inline code/__DEFINES/mode.dm:138]: Remove this line the linter is telling you about.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/civilians/support/synthetic.dm (hunk 2 above)
Comment (Drulikar) [reply code/game/jobs/job/civilians/support/synthetic.dm:59]: Likely also need to reduce `/proc/synth_slot_formula(playercount)` 's `max` argument from 2 to 1.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by realforest2001** [comment]
image
Survivor Synth and Ship Operations synth.

**Review by Drulikar — APPROVED** [review]

**Pull request merged**